circuit-overlay/src/commonMain/kotlin/com/slack/circuit/overlay/ContentWithOverlays.kt

```diff
@@ -18,17 +24,48 @@ import androidx.compose.ui.Modifier
  * @param overlayHost the [OverlayHost] to use for managing overlays.
  * @param content The regular content to render. Any overlays will be rendered over them.
  */
+@OptIn(ExperimentalAnimationApi::class)
 @Composable
 public fun ContentWithOverlays(
   modifier: Modifier = Modifier,
   overlayHost: OverlayHost = rememberOverlayHost(),
   content: @Composable () -> Unit
 ) {
   CompositionLocalProvider(LocalOverlayHost provides overlayHost) {
-    val overlayHostData by rememberUpdatedState(overlayHost.currentOverlayData)
     Box(modifier) {
       content()
-      key(overlayHostData) { overlayHostData?.let { data -> data.overlay.Content(data::finish) } }
+      val transition = updateTransition(overlayHost.currentOverlayData, label = null)
```

Comment:
I think we should keep the original code, and branch when `overlay is AnimatedOverlay`. Minimises the chances the experimental API changes causing issues.